Firestore/core/src/local/leveldb_target_cache.cc

```diff
@@ -168,7 +169,7 @@ absl::optional<TargetData> LevelDbTargetCache::GetTarget(const Target& target) {
     std::string target_key = LevelDbTargetKey::Key(row_key.target_id());
     target_iterator->Seek(target_key);
     if (!target_iterator->Valid() || target_iterator->key() != target_key) {
-      HARD_FAIL(
+      LOG_WARN(
```

Comment:
`HARD_FAIL` could not return, which effectively ensured that the loop terminated and that the `target_iterator` would be both valid and point to a record that matched `target_key`. Changing this to `LOG_WARN` allows the code to continue, breaking this condition.

I think the right thing to do is add a `continue` below this to avoid trying to parse `target_iterator->value()`.

It's probably also worth adding a unit test to validate that this works. You can do this by writing a dummy row using `LevelDbQueryTargetKey` with a canonical ID that matches the query but has an invalid target ID.

Reply:
Done.